script.js

```diff
@@ -0,0 +1,112 @@
+const minNumber = (firstNumber, secondNumber, thirdNumber, fourthNumber) => {
+    if (typeof (firstNumber && secondNumber && thirdNumber && fourthNumber) === "string") {
+      firstNumber = Number(firstNumber);
+      secondNumber = Number(secondNumber);
+      thirdNumber = Number(thirdNumber);
+      fourthNumber = Number(fourthNumber);
+    }
+    if (typeof (firstNumber && secondNumber && thirdNumber && fourthNumber) === "number") {
+      // console.log(Math.min(firstNumber, secondNumber, thirdNumber, fourthNumber));
+      return Math.min(firstNumber, secondNumber, thirdNumber, fourthNumber);
+    } else {
+      // console.log("Wrong number!");
+      return "Something went wrong! There is no minimum number!";
+    }
+  };
+  minNumber("6", 8, null, "-1");
+  
+  const greatestCommonDivider = (x, y) => {
+    if (typeof (x && y) === "string") {
+      x = Number(x);
+      y = Number(y);
+    }
+    if (typeof x !== "number" || typeof y !== "number") {
+      // console.log("Wrong action!");
+      return "Something went wrong!";
+    } else {
+      x = Math.abs(x);
+      y = Math.abs(y);
+      while (y) {
+        let t = y;
+        y = x % y;
+        x = t;
+      }
+      // console.log(x);
+      return x;
+    }
+  };
+  
+  greatestCommonDivider("8", "22");
+  
+  const isPerfect = (number1) => {
+    if (typeof num1 === "string") {
+      number1 = Number(num1);
+    }
+    if (typeof number1 === "number") {
+      let temperature = 0;
+      for (let i = 1; i <= number1 / 2; i++) {
+        if (number1 % i === 0) {
+            temperature += i;
+        }
+      }
+      if (temperature === number1 && temperature !== 0) {
```

Comment:
It can be simplified 
```suggestion
     return temperature === number1 && temperature !== 0;
```